tests/acceptance/async_test_base.py

```diff
@@ -91,11 +92,10 @@ def on_open_error(self, connection, error):
 
     def on_timeout(self):
         """called when stuck waiting for connection to close"""
-        self.logger.info('on_timeout at %s', datetime.utcnow())
-        # force the ioloop to stop
-        self.logger.debug('on_timeout called')
-        self.connection.ioloop.stop()
-        raise AssertionError('Test timed out')
```

Comment:
Assertion here doesn't help, since ioloops in some adapters, such as `LibevConnection`, suppress the exception. PR #678 will handle test timeout in a robust way, once it's merged.